python/dask_cudf/dask_cudf/backends.py

```diff
@@ -115,21 +115,14 @@ def concat_cudf(
 try:
 
     from dask.dataframe.utils import group_split_dispatch, hash_object_dispatch
-    from cudf.core.column import column, CategoricalColumn, StringColumn
+    from cudf.core.column import column
 
-    def _handle_string(s):
-        if isinstance(s._column, StringColumn):
-            s = s._hash()
-        return s
-
-    def safe_hash(df):
-        frame = df.copy(deep=False)
+    def safe_hash(frame):
+        index = frame.index
         if isinstance(frame, cudf.DataFrame):
-            for col in frame.columns:
-                frame[col] = _handle_string(frame[col])
-            return frame.hash_columns()
+            return cudf.Series(frame.hash_columns(), index=index)
         else:
-            return _handle_string(frame)
+            return cudf.Series(frame.hash_values(), index=index)
```

Comment:
Should we maintain the Series name here or is that unnecessary?

Reply:
I don't think it is necessary, the name is lost for the pandas dispatch as well.  The only reason we need to preserve the index is to avoid hiccups in dask's `assign` method.